lisa/sut_orchestrator/util/device_pool.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any, List, Optional
+
+from lisa.sut_orchestrator.util.schema import HostDevicePoolSchema, HostDevicePoolType
+from lisa.util import LisaException, SkippedException
+
+
+class BaseDevicePoolImpl:
+    def _create_device_pool(
+        self,
+        pool_type: HostDevicePoolType,
+        vendor_id: str,
+        device_id: str,
+    ) -> None:
+        raise NotImplementedError("'_create_device_pool' is not implemented")
+
+    def _get_primary_nic_id(self) -> List[str]:
+        raise NotImplementedError("'_get_primary_nic_id' is not implemented")
+
+    def _get_devices_from_pool(
```

Comment:
how about `request_devices`? It looks the method is to request a number of devices, and assign to a VM. The `get` is like a read-only method. The `from_pool` is not needed, because the class is for pool.

Reply:
@squirrelsc

Sure, will change name to `release_devices `and `request_devices `

Assignment is a bit different for libvirt/hyperv platform. Libvirt construct the XML as per need using `hostdev` and no action is needed on host to be performed before-hand whereas, for hyperv platform, we have to remove/dismount device from host and assign it to vm. So having different function would make sense and easy to maintain for any before-hand action for hyperv DDA.

Same thing applied when you are about to put device back to the pool. You again have to remove device from VM and mound/enable it on the host. Libivirt does this on his own where we have to do this explicitly for hyperv platform. Having separate function would make it easy to maintain.

Reply:
@squirrelsc Agree. But i created separate function which will be called from request/release part only. This way we can handle the assignment/removal related extra configuration properly